MvvmCross/ViewModels/Result/MvxResultViewModelManager.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MS-PL license.
+// See the LICENSE file in the project root for more information.
+
+namespace MvvmCross.ViewModels.Result;
+
+public class MvxResultViewModelManager : IMvxResultViewModelManager
+{
+    private readonly Dictionary<Type, HashSet<IMvxViewModel>> _registrations = new();
```

Comment:
Are there any implications having strong references to ViewModels in this dict?

Reply:
We need to find which VMs to notify by the result type in `MvxResultViewModelManager.SetResult`. Notice that `MvxResultViewModelManager.SetResult` takes `IMvxResultSettingViewModel` parameter while other methods take `IMvxResultAwaitingViewModel`. `IMvxResultSettingViewModel` doesn't know anything about the awaiter, and the only thing in common for them is `TResult` - that's why we need to store VMs associated to the `TResult` type.